src/views/AnnualReport.vue

```diff
@@ -431,44 +433,58 @@ export default {
   },
 
   computed: {
-    ...mapState(['currentDate', 'ARFilingYear', 'nextARDate', 'lastAgmDate', 'entityType', 'entityName',
-      'entityIncNo', 'entityFoundingDate', 'lastPreLoadFilingDate', 'directors', 'filingData']),
+    ...mapState(['currentDate', 'ARFilingYear', 'arMinDate', 'arMaxDate', 'nextARDate', 'entityType',
+      'entityName', 'entityIncNo', 'entityFoundingDate', 'directors', 'filingData']),
 
-    ...mapGetters(['isBComp', 'isCoop', 'isRoleStaff', 'isAnnualReportEditable', 'reportState', 'lastCOAFilingDate',
-      'lastCODFilingDate']),
+    ...mapGetters(['isBComp', 'isCoop', 'isRoleStaff', 'isCurrentFilingEditable', 'reportState',
+      'lastCOAFilingDate', 'lastCODFilingDate', 'currentYear']),
 
     /** Returns True if loading container should be shown, else False. */
     showLoadingContainer (): boolean {
       return (!this.dataLoaded && !this.fetchErrorDialog && !this.resumeErrorDialog &&
         !this.saveErrorDialog && !this.paymentErrorDialog)
     },
 
-    /** The As Of date, used to query data, as Effective Date, and as Annual Report Date. */
+    /**
+     * The As Of date, used to query data, as Effective Date, and as Annual Report Date.
+     * (Depends on whether entity is a Coop or BComp.)
+     */
     asOfDate (): string {
```

Comment:
Fixing the code is better than assumptions and comments.

Doing both is better still.